src/commands/deployCommands/create.js

```diff
@@ -58,10 +58,24 @@ module.exports = {
 						Promise.all(argv.versionIds.map(runE2EWithRetry))
 				)
 				.catch(error => {
-					console.log(chalk.red(`Stopping deployment: ${error}`));
-					console.log(chalk.red('Cleaning up failed deployment...'));
-					// clean up deployed versions and exit
-					return deploy.del().then(() => process.exit(1));
+					if (error.includes("already deployed")) {
+						console.log(chalk.red(`Stopping deployment: ${error}`));
+						console.log(chalk.red('Cleaning up failed deployment...'));
+
+						let exitCode = 1;
+						// clean up deployed versions and exit
+						return deploy.del().then(() => process.exit(1));
+					} else {
+						console.log(chalk.red(`Stopping deployment: ${error}`));
+						console.log(chalk.red('Cleaning up failed deployment...'));
+
+						let exitCode = 2;
+						// clean up deployed versions and exit
+						return deploy.del().then(() => process.exit(1));
+					}
```

Comment:
Hmm, it looks like this update doesn't change the behavior of the code - it assigns a local `exitCode` variable, but then doesn't use it on either side of the `if` block, and it doubles the amount of code. I _think_ that the intent is something more like

```js
.catch(error => {
  console.log(chalk.red(`Stopping deployment: ${error}`));
  console.log(chalk.red('Cleaning up failed deployment...'));

  const exitCode = error.includes("already deployed") ? 2 : 1;
  return deploy.del().then(() => process.exit(exitCode));
})
```

Reply:
yeah, you are absolutely right, initially I was thinking to use `if-else` to cover more error codes.

Reply:
I made the acceptable_error and unacceptable_error as constants as well.

Reply:
Sounds good. If/when we need to cover more error codes, we should probably break out that 'pattern matching' behavior to a separate `getErrorCode(error)` function to encapsulate the logic.